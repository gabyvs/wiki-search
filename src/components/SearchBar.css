.search-bar-icon-button {
  background-color: #292929;
}

.search-bar-icon-button:hover {
  background-color: #1f1f1f!important;
}