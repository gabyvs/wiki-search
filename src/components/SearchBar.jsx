import React  from 'react';
import styled from 'styled-components';

const wikiStyles = {
  transform: "scale(2)",
  border: "1px solid #242424"
};

const Form = styled.form`
  margin-left: 3rem;
  display: flex;
  border: 1px solid #242424;
  border-radius: 4px;
`;

const Input = styled.input`
  background-color: #292929;
  border: none;
  height: 4.4rem;
  width: 55rem;
  padding: .5rem 1rem;
  color: white;
`;

const I = styled.i`
  color: white;
`;

const IconButton = styled.button`
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 4.4rem;
  height: 4.4rem;
  border: none;
  background-color: #292929;
`;

const SearchBar = () => (
  <>
    <div><i style={wikiStyles} className="fab fa-wikipedia-w" /></div>
    <Form>
      <Input type="search" />
      <IconButton type="submit">
        <I className="fas fa-search" />
      </IconButton>
      <IconButton type="button">
        <I className="fas fa-random" />
      </IconButton>
    </Form>
  </>
);

export default SearchBar;