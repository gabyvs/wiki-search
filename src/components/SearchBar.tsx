import React  from 'react';

const styles = {
  button: {
    cursor: 'pointer',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '4.4rem',
    height: '4.4rem',
    border: 'none',
    backgroundColor: '#292929',
  },
  form: {
    marginLeft: '3rem',
    display: 'flex',
    border: '1px solid #242424',
    borderRadius: '4px'
  },
  icon: {
    color: 'white'
  },
  input: {
    backgroundColor: '#292929',
    border: 'none',
    height: '4.4rem',
    width: '55rem',
    padding: '.5rem 1rem',
    color: 'white',
  },
  logo: {
    transform: "scale(2)"
  }
};

const SearchBar = () => (
  <>
    <div><i style={styles.logo} className="fab fa-wikipedia-w" /></div>
    <form style={styles.form}>
      <input type="search" style={styles.input}/>
      <button type="submit" style={styles.button}>
        <i className="fas fa-search" style={styles.icon}/>
      </button>
      <button type="button" style={styles.button}>
        <i className="fas fa-random" style={styles.icon}/>
      </button>
    </form>
  </>
);

export default SearchBar;