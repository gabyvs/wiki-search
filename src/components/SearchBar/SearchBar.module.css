@import '../../theme.css';

.logo {
  font-weight: normal;
}

.form {
  margin-left: 3rem;
  display: flex;
  border: 1px solid var(--mat-primary);
  border-radius: 4px;
}

.input {
  background-color: var(--mat-primary);
  border: none;
  height: 4.4rem;
  width: 55rem;
  padding: .5rem 1rem;
  color: white;
  font-size: 1.6rem;
}

.button {
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 4.4rem;
  height: 4.4rem;
  border: none;
  background-color: var(--mat-primary);
}

.button:hover {
  background-color: var(--dark-accent);
}

.icon {
  color: var(--mat-primary-text);
}

@media screen and (max-width: 800px) {
  .form {
    margin: 20px auto 0;
    width: 80%
  }

}