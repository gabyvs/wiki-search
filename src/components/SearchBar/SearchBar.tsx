import React  from 'react';
import './SearchBar.css';

export interface SearchBarProps {
  searchTerm?: string;
}

const SearchBar = ({ searchTerm = '' }: SearchBarProps) => {
  return (
    <>
      <h1 title="WikiSearch">
        <i className="fab fa-wikipedia-w search-bar-logo" />
      </h1>
      <form className="search-bar-form">
        <input type="search" className="search-bar-input" value={searchTerm}/>
        <button type="submit" className="search-bar-button">
          <i className="fas fa-search search-bar-button-icon"/>
        </button>
        <button type="button" className="search-bar-button">
          <i className="fas fa-random search-bar-button-icon"/>
        </button>
      </form>
    </>
  );
};

export default SearchBar;