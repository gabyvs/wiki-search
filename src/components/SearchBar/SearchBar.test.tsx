import React                         from "react";
import {
  shallow,
  ShallowWrapper }                   from "enzyme";
import SearchBar, { SearchBarProps } from "./SearchBar";

describe("SearchBar", () => {
  let searchBar: ShallowWrapper;

  beforeEach(() => {
    searchBar = shallow(<SearchBar />);
  });

  it('renders without crashing', () => {});

  it("renders correctly", () => {
    expect(searchBar).toMatchSnapshot();
  });

  it('displays empty input when called with empty term', () => {
    expect(searchBar.find('input').props().value).toEqual('');
  });

  describe('with props', () => {
    let props: SearchBarProps;

    beforeEach(() => {
      props = { searchTerm: 'searched term' };
      searchBar = shallow(<SearchBar {...props}/>);
    });

    it('displays term in input', () => {
      expect(searchBar.find('input').props().value).toEqual(props.searchTerm);
    });
  });
});
