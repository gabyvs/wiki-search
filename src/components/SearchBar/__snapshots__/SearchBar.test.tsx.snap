// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<Fragment>
  <h1
    title="WikiSearch"
  >
    <i
      className="fab fa-wikipedia-w search-bar-logo"
    />
  </h1>
  <form
    className="search-bar-form"
  >
    <input
      className="search-bar-input"
      type="search"
      value=""
    />
    <button
      className="search-bar-button"
      type="submit"
    >
      <i
        className="fas fa-search search-bar-button-icon"
      />
    </button>
    <button
      className="search-bar-button"
      type="button"
    >
      <i
        className="fas fa-random search-bar-button-icon"
      />
    </button>
  </form>
</Fragment>
`;
